src/field_spec.c

```diff
@@ -17,10 +17,15 @@ RSValueType fieldTypeToValueType(FieldType ft) {
 }
 
 void FieldSpec_Cleanup(FieldSpec* fs) {
-  if (fs->name) {
+  // if `AS` was not used, name and path are pointing at the same string
+  if (fs->name && fs->name != fs->path) {
     rm_free(fs->name);
     fs->name = NULL;
```

Comment:
If name and path are pointing to same string, `fs->name` would not be reset to `NULL` (in line 23)?
Maybe drop the check `fs->name != fs->path` in line 21 since `free` does nothing when it receives a NULL argument?
Or move it from line 21 to before line 22?